src/task-actions.ts

```diff
@@ -339,31 +339,44 @@ async function handleContractVerification(
     const paths = [mainPath];
 
     const bodyFormData = new FormData();
-    bodyFormData.append("contract-name", path.parse(mainPath).base);
+    bodyFormData.append("compiler-version", args.compilerVersion);
+    bodyFormData.append("license", args.license || "No License (None)");
+    bodyFormData.append("account-contract", args.accountContract || "false");
 
     // Dependencies (non-main contracts) are in args.paths
     if (args.paths) {
         paths.push(...args.paths);
     }
 
-    handleMultiPartContractVerification(bodyFormData, paths, hre.config.paths.root);
-
-    await axios
-        .post(voyagerUrl, bodyFormData.getBuffer(), {
-            headers: bodyFormData.getHeaders()
-        })
-        .catch(() => {
+    const sourceRegex = new RegExp("^" + hre.config.paths?.starknetSources + "/");
+    const contractNameDefault = mainPath.replace(sourceRegex, "");
+    // If contract name is not provided, use the default
+    bodyFormData.append("contract-name", args.contractName ? args.contractName : contractNameDefault);
+    // Appends all contracts to the form data with the name "file" + index
+    handleMultiPartContractVerification(bodyFormData, paths, hre.config.paths.root, sourceRegex);
+
+    await axios({
+        method: "POST",
+        url: voyagerUrl,
+        data: bodyFormData,
+        headers: {
+            "Content-Type": `multipart/form-data; boundary=${bodyFormData.getBoundary()}`
+        }
+    })
+        .catch((err) => {
             throw new HardhatPluginError(
                 PLUGIN_NAME,
                 `\
 Could not verify the contract at address ${args.address}.
-It is hard to tell exactly what happened, but possible reasons include:
+${err.response.data.message ? err.response.data.message :
+                    `It is hard to tell exactly what happened, but possible reasons include:
```

Comment:
Do we really want to display the `err.response.data.message`? Is there a case when it can be useful? Or is it always something like `An error occurred`?

Reply:
The message contains some informative errors:
- `The license is not supported`
- `The compiler version is not supported`
- `The code does not match`
- `An error occurred` (not sure if this is that much informative)

Reply:
Ok, we can ask the Voyager team to include the list of supported values in case of invalid license or compiler.